Networking/Networking/Mapper/CustomerMapper.swift

```diff
@@ -12,8 +12,11 @@ struct CustomerMapper: Mapper {
     func map(response: Data) throws -> Customer {
         let decoder = JSONDecoder()
         decoder.userInfo = [.siteID: siteID]
-        let customer = try decoder.decode(CustomerEnvelope.self, from: response).customer
-        return customer
+        do {
+            return try decoder.decode(CustomerEnvelope.self, from: response).customer
+        } catch {
+            return try decoder.decode(Customer.self, from: response)
```

Comment:
Nice! From what I see we're using the catch block as a fallback, if we fail to decode the response with `data` we try again without the `data` CodingKey. By falling to the catch block there's an unhandled error that comes with it:

```
(lldb) po return try decoder.decode(CustomerEnvelope.self, from: response).customer
▿ DecodingError
  ▿ keyNotFound : 2 elements
    - .0 : CodingKeys(stringValue: "data", intValue: nil)
    ▿ .1 : Context
      - codingPath : 0 elements
      - debugDescription : "No value associated with key CodingKeys(stringValue: \"data\", intValue: nil) (\"data\")."
      - underlyingError : nil
```

Should we log the error anywhere, or we're not interested in this case because we throw it? Something like:
```
do {
  return try decoder.decode(CustomerEnvelope.self, from: response).customer
} catch {
  return try decoder.decode(CustomerEnvelope.self, from: response)
} catch {
  DDLog("Unable to decode Customer. Error \(error)")
}

Reply:
In our app we make mappers throw errors if the decoding fails and catch them where they are used. I understanding is that this helps with flexibility, so we can have different ways of error handling based on the context of each endpoint.